### PR TITLE
Emotemenu

### PR DESCRIPTION
**Pull request by SFFan123**
Emote menu is now generated by JS and is checing for updates every hour.

### CODE REVIEW

File: src/app/public/js/script.js [hunk 1]
```diff
@@ -1,8 +1,9 @@
 //localStorage.debug = '*';
-var socket = null;
-var focused = true;
-var unread = 0;
-var emotecheck = null;
+    var socket = null;
+	var focused = true;
+	var unread = 0;
+    var emotecheck = null;
+    var emotelist = null;
```
Comment (zaanposni) [inline src/app/public/js/script.js]: rip format

File: src/app/public/js/script.js [hunk 2]
```diff
@@ -131,25 +132,52 @@ function changeOnlineStatus(online) {
 
 function hideError() {
     $("#errorbox").fadeOut("slow");
-}
+  }
+  function addEmoteCode(emote) {
```
Comment (zaanposni) [inline src/app/public/js/script.js]: rip format

File: src/app/routes.py [hunk 3]
```diff
@@ -24,5 +25,5 @@ def send_public(path):
 
 @app.route('/api/emotes')
 def send_emotes():
-    return emotehandler.emotes
+    return jdumps(emotehandler.emotes, sort_keys=False)
```
Comment (zaanposni) [inline src/app/routes.py]: you would want to import json and use `json.dumps()` (pep8 reeee)

File: src/app/sockets.py [hunk 4]
```diff
@@ -96,15 +96,8 @@ def disconnect():
...
 
-@socketio.on('checkNewEmote')
-def check_emote():
-    SHL.output(f"Checking for new emotes: {others.new_emotes}.", "S.ON checkNewEmote")
-    if others.new_emotes:
-        emit_status({"newemote": 1})
-
-
 def emit_status(status):
-    socketio.emit('status', status, broacast=True)
```
Comment (zaanposni) [inline src/app/sockets.py]: please add a parameter to use broadcast, e.g. for status updates on join we need `broadcast=True`

Comment (SFFan123) [reply src/app/sockets.py]: Note that socketio.send() and socketio.emit() are not the same functions as the context-aware send() and emit(). Also note that in the above usage there is no client context, so broadcast=True is assumed and does not need to be specified.

Comment (SFFan123) [reply src/app/sockets.py]: https://flask-socketio.readthedocs.io/en/latest/#broadcasting

File: src/app/routes.py (hunk 3 above)
Comment (SFFan123) [reply src/app/routes.py]: why import all when you just need 1 method?

**Pull request merged**